packages/components/pie-toast/src/index.ts

```diff
@@ -1,24 +1,259 @@
-import { LitElement, html, unsafeCSS } from 'lit';
-import { RtlMixin, defineCustomElement } from '@justeattakeaway/pie-webc-core';
+import {
+    LitElement,
+    html,
+    unsafeCSS,
+    nothing,
+    type TemplateResult,
+    type PropertyValues,
+} from 'lit';
+import { property, query } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { RtlMixin, defineCustomElement, dispatchCustomEvent } from '@justeattakeaway/pie-webc-core';
+import '@justeattakeaway/pie-toast';
+import '@justeattakeaway/pie-icon-button';
+import '@justeattakeaway/pie-icons-webc/dist/IconClose.js';
+import '@justeattakeaway/pie-icons-webc/dist/IconPlaceholder.js';
+import '@justeattakeaway/pie-button';
 
 import styles from './toast.scss?inline';
-import { ToastProps } from './defs';
+import {
+    ToastProps,
+    componentSelector,
+    componentClass,
+    ActionProps,
+    ON_TOAST_CLOSE_EVENT,
+    ON_TOAST_OPEN_EVENT,
+    ON_TOAST_LEADING_ACTION_CLICK_EVENT,
+    defaultProps,
+} from './defs';
 
 // Valid values available to consumers
 export * from './defs';
 
-const componentSelector = 'pie-toast';
-
 /**
  * @tagname pie-toast
  */
 export class PieToast extends RtlMixin(LitElement) implements ToastProps {
-    render () {
-        return html`<h1 data-test-id="pie-toast">Hello world!</h1>`;
-    }
+    @property()
+    public message: ToastProps['message'] = '';
```

Comment:
More of a question but should we escape/sanitize the message string?

Reply:
I don't think we need to scape it. I tested here and seems like LIT is already doing the heavy lifting for us.
<img width="902" alt="Screenshot 2024-07-19 at 12 56 07 PM" src="https://github.com/user-attachments/assets/58a0d3a1-bc85-4d0d-9d05-27d4d3b5c0bf">